### PR TITLE
chore: unify the way to create RouteDestination

### LINKED ISSUE #786: ir.RouteDestination weight  ?
*Description*:

```go
type RouteDestination struct {
	// Host refers to the FQDN or IP address of the backend service.
	Host string
	// Port on the service to forward the request to.
	Port uint32
	// Weight associated with this destination.
	// Note: Weight is not used in UDP route.
	Weight uint32
}
```

In fact, there are 2 cases, 0 and nil
-  Weight 0, which is actually a legal input
-  Weight == nil ，no setting required

### PR DESCRIPTION
**Pull request by muyuan0**
Fix #786 
When taking a look at issue 786, I found that different places use different way to create RouteDestination.
This commit unifies the behavior.

### CODE REVIEW

**Comment by muyuan0** [comment]
I would like to put each field into private, but it will break the yaml marshal used in the test (in/out diff). However, we can solve it if this type of refactoring is acceptable.

File: internal/gatewayapi/route.go [hunk 1]
```diff
@@ -684,10 +684,12 @@ func (t *Translator) processUDPRouteParentRefs(udpRoute *UDPRouteContext, resour
 		}
 
 		// weight is not used in udp route destinations
-		routeDestinations = append(routeDestinations, &ir.RouteDestination{
-			Host: service.Spec.ClusterIP,
-			Port: uint32(*backendRef.Port),
-		})
+		weight := uint32(0)
```
Comment (arkodg) [inline internal/gatewayapi/route.go:687]: lets rm this, since it wasn't part of the orig logic

File: internal/gatewayapi/route.go [hunk 2]
```diff
@@ -809,10 +811,12 @@ func (t *Translator) processTCPRouteParentRefs(tcpRoute *TCPRouteContext, resour
 		}
 
 		// weight is not used in tcp route destinations
-		routeDestinations = append(routeDestinations, &ir.RouteDestination{
-			Host: service.Spec.ClusterIP,
-			Port: uint32(*backendRef.Port),
-		})
+		weight := uint32(0)
```
Comment (arkodg) [inline internal/gatewayapi/route.go:814]: same comment as above

**Comment by arkodg** [comment]
thanks @muyuan0 , added some comments

File: internal/ir/xds.go [hunk 3]
```diff
@@ -393,14 +394,24 @@ func (r RouteDestination) Validate() error {
...
+		Port: port,
 	}
 }
 
+func NewRouteDestWithWeight(host string, port uint32, weight uint32) *RouteDestination {
+	rd := &RouteDestination{
+		Host: host,
+		Port: port,
+	}
+	if weight != 0 {
```
Comment (arkodg) [inline internal/ir/xds.go:409]: imo this check isn't needed, what if the caller wants to explicitly set the weight to zero ?

Comment (muyuan0) [reply internal/ir/xds.go]: Addressed in the new commit.

**Review by arkodg — APPROVED** [review]
LGTM, thanks !

**Pull request merged**